rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1252,6 +1258,193 @@ def __init__(
 
         self._prepare_layers()
 
+    def adjust_sparse_layers(
+        self,
+        data_example: Dict[Text, List[FeatureArray]],
+        new_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+        old_feature_sizes: Dict[Text, Dict[Text, List[int]]],
+    ) -> None:
+        """Adjusts sizes of DenseForSparse layers.
+
+        Updates sizes of DenseForSparse layers by comparing current sparse feature
+        sizes to old ones. This must be done before fine-tuning starts to account for
+        sparse features that might have changed. If any layer is changed, compiles
+        the model and fits a sample data on it to activate updated layer(s).
+
+        Args:
+            data_example: a data example that is stored in DIETClassifier class.
+            new_feature_sizes: sizes of current sparse features.
+            old_feature_sizes: sizes of sparse features the model was trained on before.
+        """
+        model_changed = False
+        for attribute in new_feature_sizes:
+            layer_dicts = self._get_dense_for_sparse_layers(attribute)
```

Comment:
A variable name shouldn't mention the data structure. A better name for `layer_dicts` will be `attribute_layers`.